.github/workflows/main.yml

```diff
@@ -0,0 +1,145 @@
+name: Pull Request Tests
+
+on:
+  push:
+    branches:
+    - feature/ContInteg
+  pull_request:
+    branches:
+    - feature/ContInteg
+
+jobs:
+  setup:
+    name: Set up
+    runs-on: ubuntu-latest
+
+    outputs:
+      tn: ${{ steps.parse.outputs.tn }}
+      bld: ${{ steps.parse.outputs.bld }}
+      test: ${{ steps.parse.outputs.test }}
+      img: ${{ steps.parse.outputs.img }}
+
+    steps:
+    - name: Checkout codes
+      uses: actions/checkout@v2
+
+    - name: Parse cases
```

Comment:
Why is all this parsing of ci.test file done in action instead of in some script that action just runs. The action file should not contain any logic about how tests are done and what they are actual doing. It's (action) purpose should be just to runs some script based on some repository event.

Reply:
I agree, most of this could be encapsulated in some script (in whatever preferred scripting language) to simplify these `.yml` files.